src/design/ConfigStrap.stanza

```diff
@@ -0,0 +1,224 @@
+then structure the array according to some heuristic.
+<DOC>
+public defmulti pack-array (ap:ArrayPacking, hi-comps:JITXObject, lo-comps:JITXObject)
+
+doc: \<DOC>
+Longitudinal Packing for Configuration Component Arrays
+
+This packing type will place the components in the
+following structure:
+
+  H0  H1  H2  ...
+  L0  L1  L2
+
+This will typically place the VDD rail on one side and ground rail
+on the other for easier connecting.
+
+<DOC>
+public defstruct LongitudinalPacking <: ArrayPacking:
+  doc: \<DOC>
+  Set the extra distance between components in the packing array.
+  <DOC>
+  margin:Double with:
+    default => 0.0
+    ensure => ensure-non-negative!
+with:
+  keyword-constructor => true
+
+
+defn is-pad-L2R? (comp:Instantiable) -> True|False :
+  val lp = landpattern(comp)
+  val p1 = get-pad-by-name!(lp, "p[1]")
+  val p2 = get-pad-by-name!(lp, "p[2]")
+
+  val p1-loc = center $ pose(p1)
+  val p2-loc = center $ pose(p2)
+
+  x(p1-loc) < x(p2-loc)
+
+defn is-pad-T2B? (comp:Instantiable) -> True|False :
+  val lp = landpattern(comp)
+  val p1 = get-pad-by-name!(lp, "p[1]")
+  val p2 = get-pad-by-name!(lp, "p[2]")
+
+  val p1-loc = center $ pose(p1)
+  val p2-loc = center $ pose(p2)
+
+  y(p1-loc) > y(p2-loc)
+
+
+defmethod pack-array (ap:LongitudinalPacking, hi-comps:JITXObject, lo-comps:JITXObject):
+  inside pcb-module:
+    val h-type = get-instantiable(hi-comps)
+    val hd = get-courtyard-dims $ h-type
+    val ld = get-courtyard-dims $ get-instantiable(lo-comps)
+
+    ; We use the aspect ratio of the part to determine whether the component
+    ; grid needs to be oriented in the X direction or the Y direction.
+    val is-Y-aligned = x(hd) < y(hd)
+
+    if is-Y-aligned:
+      val is-t2b? = is-pad-T2B?(h-type)
+      val [t-comp, b-comp] = if is-pad-T2B?(h-type):
+        [hi-comps, lo-comps]
+      else:
+        [lo-comps, hi-comps]
+
+      val w = max(x(hd), x(ld))
+      val m2 = margin(ap) / 2.0
+
+      for i in 0 to length(hi-comps) do:
+        val x-p = (w + margin(ap)) * to-double(i)
+        val y-h = (y(hd) / 2.0) + m2
+        place(t-comp[i]) at loc(x-p, y-h) on Top
+        val y-l = (y(ld) / 2.0) + m2
+        place(b-comp[i]) at loc(x-p, (- y-l)) on Top
+    else:
+      ; We are X aligned
+      val [l-comp, r-comp] = if is-pad-L2R?(h-type):
+        [hi-comps, lo-comps]
+      else:
+        [lo-comps, hi-comps]
+
+      val h = max(y(hd), y(ld))
+      val m2 = margin(ap) / 2.0
+
+      for i in 0 to length(hi-comps) do:
+        val y-p = (h + margin(ap)) * to-double(i)
+        val x-h = (x(hd) / 2.0) + m2
+        place(r-comp[i]) at loc(x-h, y-p) on Top
+        val x-l = (x(ld) / 2.0) + m2
+        place(l-comp[i]) at loc((- x-l), y-p) on Top
+
+
+doc: \<DOC>
+Configuration Bus Bundle
+
+This bundle is what is used to access the
+pin assignment supports for the ConfigStrap module.
+<DOC>
+public pcb-bundle config-strap-bus (num-elems:Int) :
+  port config : pin[num-elems]
```

Comment:
You're coding a bundle in the designs folder. No real issue except for bookkeeping (bundles versus design). Ok for you?

Reply:
The bundle is rather design-specific though. So it makes sense.

Reply:
Yes - This bundle is only applicable in this application. We do a similar thing with the bundles in the protocols directory where a bundle is only applicable to that application. 

I put it in `design` because it didn't seem to fit anywhere else. We could maybe make a `generators` or `circuits` directory or something ?

Reply:
I moved it to `circuits` but kept the structure.